app/models/payment/go_cardless/transaction.rb

```diff
@@ -81,8 +81,10 @@ def publish_failed_subscription_charge
     ChampaignQueue.push({
       type: 'subscription-payment',
       params: {
+        created_at: created_at,
         recurring_id: subscription.go_cardless_id,
-        success: 0
+        success: 0,
+        status: 'failed'
```

Comment:
Just to add - generally I think we could spend some time moving concerns away to the worker from Champaign, but I also think it's worth considering what should be in the worker and what should actually be in the gem, as well.